drivers/soundwire/intel.c

```diff
@@ -1005,9 +1005,13 @@ static int intel_trigger(struct snd_pcm_substream *substream, int cmd, struct sn
 {
 	struct sdw_cdns *cdns = snd_soc_dai_get_drvdata(dai);
 	struct sdw_intel *sdw = cdns_to_intel(cdns);
+	struct sdw_intel_link_res *res = sdw->link_res;
 	struct sdw_cdns_dma_data *dma;
 	int ret = 0;
 
+	if (res->ops && res->ops->trigger)
+		res->ops->trigger(substream, cmd, dai);
```

Comment:
Should this be done after the `snd_soc_dai_get_dma_data()` ?

Reply:
> Should this be done after the `snd_soc_dai_get_dma_data()` ?

Given that 1. the trigger ops doesn't need the dma data, and 2. the trigger ops is used to send IPC to firmware. I think we should call the trigger ops no matter we get dma data successfully or not.